plugins/webhooks/index.js

```diff
@@ -1156,92 +62,27 @@ const webhooksPlugin = {
-                            const commitAuthors =
-                                Array.isArray(parsed.commitAuthors) && parsed.commitAuthors.length !== 0
-                                    ? parsed.commitAuthors
-                                    : [username];
-                            const validCommitAuthors = commitAuthors.filter(author => !ignoreUser.includes(author));
-
-                            if (!validCommitAuthors.length) {
-                                message = `Skipping because user ${username} is ignored`;
-                                request.log(['webhook', hookId], message);
-
-                                return h.response({ message }).code(204);
-                            }
-                        }
-
-                        const token = await obtainScmToken({ pluginOptions, userFactory, username, scmContext, scm });
-
-                        if (action !== 'release' && action !== 'tag') {
-                            let scmUri;
-
-                            if (type === 'pr') {
-                                scmUri = await scm.parseUrl({ checkoutUrl, token, scmContext });
-                            }
-                            parsed.changedFiles = await scm.getChangedFiles({
-                                payload: request.payload,
-                                type,
-                                token,
-                                scmContext,
-                                scmUri,
-                                prNum
-                            });
-                            request.log(['webhook', hookId], `Changed files are ${parsed.changedFiles}`);
-                        } else {
-                            // The payload has no sha when webhook event is tag or release, so we need to get it.
-                            try {
-                                parsed.sha = await getCommitRefSha({
-                                    scm,
-                                    token,
-                                    ref,
-                                    refType: 'tags',
-                                    checkoutUrl,
-                                    scmContext
-                                });
-                            } catch (err) {
-                                request.log(['webhook', hookId, 'getCommitRefSha'], err);
-
-                                // there is a possibility of scm.getCommitRefSha() is not implemented yet
-                                return h.response({ message }).code(204);
-                            }
-                        }
-
-                        if (type === 'pr') {
-                            // disregard skip ci for pull request events
-                            return pullRequestEvent(pluginOptions, request, h, parsed, token);
-                        }
+                        return await startHookEvent(pluginOptions, request, h, parsed);
```

Comment:
I don't know well about scope of this PR. Can we start event directly instead of calling queue service API when `/webhook` is called?

Reply:
Yes. The scope of this PR is to implement commonly referenced helpers.
The process of calling the quese-service API will be implemented in another context.